### ISSUE TITLE
🧠 URGENT: COACH Universal Training - 20 Cryptos × 4 Timeframes

### ISSUE THREAD

**Issue by papsiei** [papsiei/trade-machine#5]
## 🎯 COACH Universal Training Complete

**Priority:** ⭐⭐⭐⭐⭐ URGENT  
**Status:** ⏳ IN PROGRESS  
**Timeline:** 30 minutes - 2 days  
**Part of:** PHASE 0 - Foundation Hardening (#2)

### Objective
Complete COACH Universal training with 5M samples (20 cryptos × 4 timeframes)

### Background
```
Current State:
├─ Database: 56.6M rows ready (Parquet optimized)
├─ Features: 41 technical indicators implemented
├─ Script: scripts/coach/fase1-top20-completo.py (ready)
└─ Expected accuracy: 73-75% (baseline multi-asset)

Previous (BTC only):
├─ 84.5% accuracy (1 crypto, 1 timeframe)
└─ 71k samples
```

#### 1. Execute Training (30 min - 1 hour)
```bash
cd c:\Users\cuaqu\Desktop\trade-clean
.venv312\Scripts\python.exe scripts\coach\fase1-top20-completo.py
```

**What it does:**
- Loads 20 cryptos × 4 timeframes = 80 datasets
- Calculates 41 features per dataset
- Generates BULL/NOT-BULL labels
- Trains Random Forest (200 trees)
- Saves model: `models/coach_universal_fase1.pkl`

#### 2. Analyze Results (2-3 hours)

**Per-Crypto Breakdown:**
- [ ] Accuracy BTC: Expected 78-82% (best, most data)
- [ ] Accuracy ETH: Expected 75-78%
- [ ] Accuracy Top 10: Expected 75-80%
- [ ] Accuracy Top 20: Expected 70-75%
- [ ] Identify: Best/worst cryptos for COACH

**Per-Timeframe Breakdown:**
- [ ] Accuracy 15m: Expected ??%
- [ ] Accuracy 1h: Expected ??%
- [ ] Accuracy 4h: Expected ??%
- [ ] Accuracy 1d: Expected ??%
- [ ] Identify: Best/worst timeframes

**Feature Importance:**
- [ ] TOP 15 features identified
- [ ] What COACH learns most from (RSI, MACD, etc.)
- [ ] Crypto-specific differences (BTC vs SOL)
- [ ] Timeframe-specific differences (15m vs 1d)

#### 3. Validation (1-2 hours)
- [ ] Cross-validation: 5-fold Purged K-Fold
- [ ] Walk-Forward: Train 2020-2023, Test 2024
- [ ] Overfitting check: CV ≈ WF (gap <3pp expected)
- [ ] Generalization confirmed

### Success Criteria

**GO (Proceed to optimization):**
```
✅ Accuracy overall: 73-75%
✅ Accuracy BTC: ≥78%
✅ Accuracy Top 10: ≥75%
✅ CV vs WF gap: <3pp
```

**REVIEW (Needs analysis):**
```
⚠️ Accuracy: 70-73%
⚠️ Some cryptos <70%
⚠️ CV vs WF gap: 3-5pp
```

**NO-GO (Major issues):**
```
❌ Accuracy: <70%
❌ Most cryptos failing
❌ CV vs WF gap: >5pp (overfitting)
```

### Expected Output

```
Dataset universal: ~5-6 million samples
Accuracy esperada: 73-75%
Feature importance: TOP features discovered
Per crypto: What COACH learns from BTC vs ETH vs SOL
Per timeframe: Differences 15m vs 1h vs 4h vs 1d

Files generated:
├─ models/coach_universal_fase1.pkl (6-10 MB)
├─ results/fase1-coach-universal.json
└─ reports/fase1-feature-importance.txt
```

### Next Steps

**If GO:**
→ Issue #5: Hyperparameter Optimization  
→ Issue #6: Threshold Optimization

**If REVIEW:**
→ Analyze which cryptos/timeframes fail  
→ Feature engineering adjustments

**If NO-GO:**
→ Deep dive on data quality  
→ Review feature calculation

---

**Related Issues:** #2 (PHASE 0)  
**Blocks:** #5, #6, #3, #4  
**Script:** `scripts/coach/fase1-top20-completo.py`
